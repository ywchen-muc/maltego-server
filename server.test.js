const request = require('supertest');
const { app, server } = require('./server');

describe('TEST server.js', () => {
    afterAll(async () => {
        server.close();
      });

    it("GET / route , should return graphs", async () => {
        return request(app)
          .get("/")
          .expect(200)
          .then(res => {
            expect(res.body).toMatchSnapshot();
          });
    });
    
    it("GET /:id , should return graph where id=graph_1", async () => {
        const id = "grph_1";
        return request(app)
          .get(`/${id}`)
          .expect(200)
          .then(res => {
            expect(res.body).toMatchSnapshot();
          });
    });
    
    it("GET /:id , should return error message when graph with the specified id is not found", async () => {
        const id = 10;
        return request(app)
          .get(`/${id}`)
          .expect(400)
          .then(res => {
            expect(res.body).toMatchSnapshot();
          });
    });

    it("GET /node:searchfield , should return graph where contains node id=serachfield", async () => {
        const searchfield = "Node 3";
        return request(app)
          .get(`/node/${searchfield}`)
          .expect(200)
          .then(res => {
            expect(res.body).toMatchSnapshot();
          });
    });

    it("DELETE /:id , should return success message when id=graph_1 deleted", async () => {
        const id = "grph_1";
        return request(app)
          .delete(`/${id}`)
          .expect(200)
          .then(res => {
            expect(res.body).toMatchSnapshot();
          });
    });

    it("DELETE /:id , should return error message when graph with the specified id is not found", async () => {
        const id = 10;
        return request(app)
          .delete(`/${id}`)
          .expect(400)
          .then(res => {
            expect(res.body).toMatchSnapshot();
          });
    });

    it("POST / , should return the successfully inserted new graph object", async () => {
        const newGraph = {
            id: '',
            name: 'Grph_27',
            data: {}
        };

        return request(app)
            .post("/")
            .send(newGraph)
            .expect(200)
            .then(res => {
            expect(res.body).toMatchSnapshot();
            });
    });

    it("POST / , should return error if name is null", async () => {
        const newGraph = {
            id: '',
            name: null,
            data: {}
        };

        return request(app)
            .post("/")
            .send(newGraph)
            .expect(400)
            .then(res => {
            expect(res.body).toMatchSnapshot();
        });
    });
});