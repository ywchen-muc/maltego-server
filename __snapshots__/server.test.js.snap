// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST server.js DELETE /:id , should return error message when graph with the specified id is not found 1`] = `
Object {
  "error": "Graph not found",
}
`;

exports[`TEST server.js DELETE /:id , should return success message when id=graph_1 deleted 1`] = `
Object {
  "success": "Graph deleted!",
}
`;

exports[`TEST server.js GET / route , should return graphs 1`] = `
Array [
  Object {
    "data": Object {
      "edges": Array [
        Object {
          "source": "nd_1",
          "target": "nd_2",
        },
        Object {
          "source": "nd_1",
          "target": "nd_3",
        },
        Object {
          "source": "nd_1",
          "target": "nd_4",
        },
      ],
      "nodes": Array [
        Object {
          "id": "nd_1",
          "label": "Node 1",
        },
        Object {
          "id": "nd_2",
          "label": "Node 2",
        },
        Object {
          "id": "nd_3",
          "label": "Node 3",
        },
        Object {
          "id": "nd_4",
          "label": "Node 4",
        },
      ],
    },
    "id": "grph_1",
    "name": "Graph 1",
  },
  Object {
    "data": Object {
      "edges": Array [
        Object {
          "source": "nd_1",
          "target": "nd_2",
        },
      ],
      "nodes": Array [
        Object {
          "id": "nd_1",
          "label": "Node 1",
        },
        Object {
          "id": "nd_2",
          "label": "Node 2",
        },
      ],
    },
    "id": "grph_2",
    "name": "Graph 2",
  },
  Object {
    "data": Object {
      "edges": Array [
        Object {
          "source": "nd_1",
          "target": "nd_2",
        },
        Object {
          "source": "nd_1",
          "target": "nd_3",
        },
        Object {
          "source": "nd_1",
          "target": "nd_4",
        },
        Object {
          "source": "nd_1",
          "target": "nd_5",
        },
      ],
      "nodes": Array [
        Object {
          "id": "nd_1",
          "label": "Node 1",
        },
        Object {
          "id": "nd_2",
          "label": "Node 2",
        },
        Object {
          "id": "nd_3",
          "label": "Node 3",
        },
        Object {
          "id": "nd_4",
          "label": "Node 4",
        },
        Object {
          "id": "nd_5",
          "label": "Node 5",
        },
        Object {
          "id": "nd_6",
          "label": "Node 6",
        },
      ],
    },
    "id": "grph_3",
    "name": "Graph 3",
  },
]
`;

exports[`TEST server.js GET /:id , should return error message when graph with the specified id is not found 1`] = `"Not found"`;

exports[`TEST server.js GET /:id , should return graph where id=graph_1 1`] = `
Object {
  "data": Object {
    "edges": Array [
      Object {
        "source": "nd_1",
        "target": "nd_2",
      },
      Object {
        "source": "nd_1",
        "target": "nd_3",
      },
      Object {
        "source": "nd_1",
        "target": "nd_4",
      },
    ],
    "nodes": Array [
      Object {
        "id": "nd_1",
        "label": "Node 1",
      },
      Object {
        "id": "nd_2",
        "label": "Node 2",
      },
      Object {
        "id": "nd_3",
        "label": "Node 3",
      },
      Object {
        "id": "nd_4",
        "label": "Node 4",
      },
    ],
  },
  "id": "grph_1",
  "name": "Graph 1",
}
`;

exports[`TEST server.js POST / , should return error if name is null 1`] = `
Object {
  "error": "Invalid params",
}
`;

exports[`TEST server.js POST / , should return the successfully inserted new graph object 1`] = `
Object {
  "data": Object {},
  "id": "grph_27",
  "name": "Grph_27",
}
`;
